### DISCUSSION TITLE
Absolute vs. relative paths in RPMs

### DISCUSSION THREAD

**Discussion by carlosabalde** [goreleaser/nfpm#828] [Q&A]
In our transition from FPM to nFPM, we've noticed this difference (running the command as a regular user) comparing the generated packages:

- FPM
  ```
  ❯ rpm2cpio ./vcs-connector-2.2.5-1.rhel9.x86_64.rpm | cpio -idmv
  ...
  ./lib/systemd/system/vcs-connector.service
  ./usr/bin/vcs-connector
  ...
  ```

- nFPM
  ```
  ❯ rpm2cpio ./vcs-connector-2.2.5-1.rhel9.x86_64.rpm | cpio -idmv
  ...
  cpio: /lib/systemd/system/vcs-connector.service: Cannot open: Permission denied
  /lib/systemd/system/vcs-connector.service
  cpio: /usr: Cannot change mode to rwxr-xr-x: Operation not permitted
  cpio: /usr: Cannot utime: Operation not permitted
  /usr
  cpio: /usr/bin: Cannot change mode to rwxr-xr-x: Operation not permitted
  cpio: /usr/bin: Cannot utime: Operation not permitted
  /usr/bin
  cpio: /usr/bin/vcs-connector: Cannot open: Permission denied
  /usr/bin/vcs-connector
  ...
  ```

In summary, paths generated by nFPM seem to be absolute, and the ones generated by FPM are relative. I've checked other packages like https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm and in there they are relative too. I've tried to use relative paths in `contents.dst` but nothing changes. No idea if this is a relevant difference or not. Anybody knows? :)

**Comment by caarlos0** [comment]
if I recall correctly, they should be absolute, or at least I think that's what rpmbuild does.

I might be wrong though, it has been a while.

Generally, I would guess that either way it is not a problem, as packages generated by both fpm and nfpm have been used for years now 🤔

**Reply by carlosabalde** [reply]
Yes, as far as I know, it's not a problem, no matter if they are absolute or relative paths. It's just a little bit inconvenient and apparently different from the usual practice, and that's why I asked, just in case it wasn't intentional :)
